File: c++_code(dirty)/Particle.cpp
#include "Particle.h"

Particle::Particle(std::vector<Vel_pos> domain, std::vector<Vel_pos> u,int t,int c,bool f):_t(t),_c(c)
{
    _u=u;
    _domain=domain;
     _f=f;
    int size=_domain[0].get_size();
    double* arg_tab=new double[size];
    for (int i=0; i<size; i++){
        arg_tab[i]=((double)rand()/(double)RAND_MAX)*(domain[1][i]-domain[0][i])+domain[0][i];
    }
    _arg=Vel_pos(arg_tab,size);
    _vel=Vel_pos(0.0,size);
    if(f==false)
    	_cur_fit=fit_func(_u,_t,_c,_arg);
    if(f==true)
	_cur_fit=fit_func_fcm(_u,_t,_c,_arg);
    _best_fit=_cur_fit;
    _best_arg=_arg;
    delete [] arg_tab;
}

Particle::Particle(std::vector<Vel_pos>  domain,std::vector<Vel_pos>  u, Vel_pos arg,int t,int c,bool f):_arg(arg),_best_arg(arg),_t(t),_c(c)
{
    _domain=domain;
    _u=u;
    _f=f;
    int size=domain[0].get_size();
    _vel=Vel_pos(0.0,size);
    _cur_fit=0;
    _best_fit=0;
}

Particle::~Particle()
{

}

void Particle::fit(Vel_pos v){
    _vel=v;
    _arg=_arg+v;
    if(_f==false){
    	_cur_fit=fit_func(_u,_t,_c,_arg);
    	if(_cur_fit<=_best_fit){
        	_best_fit=_cur_fit;
        	_best_arg=_arg;
    	}
    }
    if(_f==true){
    	_cur_fit=fit_func_fcm(_u,_t,_c,_arg);
    	if(_cur_fit<=_best_fit){
        	_best_fit=_cur_fit;
        	_best_arg=_arg;
    	}
    }
}

File: Matlab_code_with_GUI/Particle.m
classdef Particle < handle
    properties
        vel;
        arg;
        best_fit;
        best_arg;
        cur_fit;
        domain;
        u;
        t;
        c;
    end
    methods
        
        function obj=Particle(domain,u,t,c)
            obj.t=t;
            obj.c=c;
            obj.u=u;
            ds=size(domain);
            ds=ds(2);
            obj.arg=(rand*(domain(2,1)-domain(1,1))+domain(1,1));
            obj.vel=0;
            for i=2:ds
                obj.arg=[obj.arg (rand*(domain(2,i)-domain(1,i))+domain(1,i))];
                obj.vel=[obj.vel 0];
            end
            obj.cur_fit=fit_func(u,t,c,obj.arg);
            obj.best_fit=obj.cur_fit;
            obj.best_arg=obj.arg;
            
        end
     
        
        function Particle=fit(Particle,v)
            Particle.vel=v;
            Particle.arg=Particle.arg+v;
            Particle.cur_fit=fit_func(Particle.u,Particle.t,Particle.c,Particle.arg);
            if(Particle.cur_fit<=Particle.best_fit)
                Particle.best_fit=Particle.cur_fit;
                Particle.best_arg=Particle.arg;
            end
        end
    end
        
end